mockgen/parse.go

```diff
@@ -409,8 +411,16 @@ func (p *fileParser) parseType(pkg string, typ ast.Expr) (model.Type, error) {
 			case (*ast.BasicLit):
 				value = val.Value
 			case (*ast.Ident):
-				// when the length is a const
+				// when the length is a const defined locally
 				value = val.Obj.Decl.(*ast.ValueSpec).Values[0].(*ast.BasicLit).Value
+			case (*ast.SelectorExpr):
+				// when the length is a const defined in an external package
+				usedPkg, err := importer.Default().Import(fmt.Sprintf("%s", val.X))
```

Comment:
nit: this error should be checked
nit: instead of `fmt.Sprintf("%s", val.X)` could you use val.X.Name

Reply:
> nit: this error should be checked

Thanks for the catch! I missed this error.
https://github.com/golang/mock/pull/569/commits/d272a9382fc5b18efa25a2caa70b85b80435b839

> nit: instead of fmt.Sprintf("%s", val.X) could you use val.X.Name

The `X` field in the `SelectorExpr` type doesn't seem to have "Name" as its type is just the `Expr` interface.
https://github.com/golang/mock/pull/569/files

Reply:
Sorry I should have been more clear, you are right. To get at name you must cast it. val.X should always be an identifier in this case so `val.X.(*ast.Ident).Name` should work. But this is good to, less panic prone should ast representation change I suppose.

Reply:
Ah, I see. Your way looks fine too, but I feel using `fmt.Sprintf` is less panic prone as you say.
Thanks for the clarification 😄